src/test/groovy/org/prebid/server/functional/tests/AmpSpec.groovy

```diff
@@ -266,4 +266,32 @@ class AmpSpec extends BaseSpec {
         assert bidderRequest.imp[0]?.banner?.format[0]?.w == ampStoredRequest.imp[0].banner.format[0].w
         assert bidderRequest.regs?.ext?.gdpr == ampStoredRequest.regs.ext.gdpr
     }
+
+    def "PBS should generate UUID for BidRequest id and merge StoredRequest when generate-storedrequest-bidrequest-id = true or id = {{UUID}}"() {
+        given: "Pbs config with settings.generate-storedrequest-bidrequest-id and default-account-config"
+        def pbsService = pbsServiceFactory.getService(["settings.generate-storedrequest-bidrequest-id": (genarateBidRequest)]);
+
+        and: "Default AMP request with custom Id"
+        def ampRequest = AmpRequest.defaultAmpRequest.tap {
+            tagId = bidRequestId
+        }
+
+        and: "Default BidRequest"
+        def ampStoredRequest = BidRequest.defaultBidRequest
+
+        and: "Stored request in DB"
+        def storedRequest = StoredRequest.getDbStoredRequest(ampRequest, ampStoredRequest)
+        storedRequestDao.save(storedRequest)
+
+        when: "PBS processes amp request"
+        def ampResponse = pbsService.sendAmpRequest(ampRequest)
+
+        then: "AmpResponse and BidRequest shouldn't equals id"
+        assert ampResponse.ext?.debug?.resolvedRequest?.getId() != bidRequestId
```

Comment:
1) Please use native Groovy property accessors.
2) Please check actual bidder call ID instead of or in addition to debug info.